### PR TITLE
Upgraded Spring to 5.2.22 for 6.1

### PR DESCRIPTION
**Pull request by MarekB01**
**A Brief Overview**
Upgraded Spring to 5.2.22 , Boot to 2.3.12 and security to 5.4.10. 

The upgrade required additional changes which were done for 6.2. Cherry picked these changes:

- https://github.com/BroadleafCommerce/BroadleafCommerce/pull/2448
- https://github.com/BroadleafCommerce/BroadleafCommerce/commit/e0a07c787ccffce7e540ef073ef973fa81d0c14a
- https://github.com/BroadleafCommerce/BroadleafCommerce/pull/2675

This will also require RELEASE NOTES explaining removal of deprecated annotations and example of migrating to new one.

**Link to QA**
[QA-4695](https://github.com/BroadleafCommerce/QA/issues/4695)

### CODE REVIEW

**Review by oleksiimiroshnyk — APPROVED** [review]

**Comment by riteshadhikari17** [comment]
@MarekB01 I doubt major spring upgrade would be this easy. Have you tested with this? I see build failure in jenkins as well as locally. Run `mvn clean install` which should run unit tests. After that builds successfully, u need to run Refsite as well to verify everything is working fine.

**Comment by riteshadhikari17** [comment]
Also let me tell u more about labels. When gathering release notes, these labels are used. So any PR can be either `Bug`, `Enhancement`, `Feature` or `Security`. So pick only one of those. Severity only applies if its a bug and it should be either `Critical`,`Major `or`Minor`. Even though QA has more options, PRs should only have one of those three.

**Comment by riteshadhikari17** [comment]
Seems link spring-5.2 is not compatible with springboot-2.1. From release notes, I see [spboot-2.2](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.2-Release-Notes) was updated with sp-5.2 and spboot-2.4 with sp-5.3. So I guess test with latest version of [spboot-2.3](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.3-Release-Notes) which is [2.3.12](https://mvnrepository.com/artifact/org.springframework.boot/spring-boot/2.3.12.RELEASE).

File: admin/broadleaf-contentmanagement-module/src/main/java/org/broadleafcommerce/cms/page/domain/PageImpl.java [hunk 1]
```diff
@@ -74,19 +75,59 @@
...
-        @AdminPresentationOverride(name="pageTemplate.templateDescription", value=@AdminPresentation(excluded = true)),
-        @AdminPresentationOverride(name="pageTemplate.templateName", value=@AdminPresentation(excluded = true)),
-        @AdminPresentationOverride(name="pageTemplate.locale", value=@AdminPresentation(excluded = true))
-    }
-)
+@AdminPresentationMergeOverrides(value = { 
+    @AdminPresentationMergeOverride(name = "auditable.createdBy.id",
+        mergeEntries = { 
+            @AdminPresentationMergeEntry(propertyType = PropertyType.AdminPresentation.READONLY, booleanOverrideValue = true),
+            @AdminPresentationMergeEntry(propertyType = PropertyType.AdminPresentation.VISIBILITY, overrideValue = "HIDDEN_ALL")
```
Comment (danielcolgrove) [inline admin/broadleaf-contentmanagement-module/src/main/java/org/broadleafcommerce/cms/page/domain/PageImpl.java:82]: These literal "HIDDEN_ALL" should probably use VisibilityEnum.HIDDEN_ALL

**Review by danielcolgrove — CHANGES_REQUESTED** [review]
It would probably worth showcasing a before and after migration of an AdminPresentationOverride vs. an AdminPresentationMergeOverride in the release notes.

File: admin/broadleaf-contentmanagement-module/src/main/java/org/broadleafcommerce/cms/page/domain/PageImpl.java (hunk 1 above)
Comment (riteshadhikari17) [reply admin/broadleaf-contentmanagement-module/src/main/java/org/broadleafcommerce/cms/page/domain/PageImpl.java:82]: Required type for `overrideValue` is String so it does not accept enum.
image

Comment (riteshadhikari17) [reply admin/broadleaf-contentmanagement-module/src/main/java/org/broadleafcommerce/cms/page/domain/PageImpl.java:82]: We can introduce `overrideVisibility` and update everywhere. But that will have to be separate ticket. This is how its set right now in `BasicFieldMetadataProvider`
```
} else if (entry.getKey().equals(PropertyType.AdminPresentation.VISIBILITY)) {
      fieldMetadataOverride.setVisibility(VisibilityEnum.valueOf(stringValue));
```

**Review by danielcolgrove — APPROVED** [review]

**Pull request merged**